File: AspNetMvcUnleashed/0672329980_source/Chapter11/CS/MvcApplication1/MvcApplication1/Controllers/HomeController.cs
using System.Linq;
using System.Web.Mvc;
using MvcApplication1.Models;
using MvcApplication1.CutomActionFilters;

namespace MvcApplication1.Controllers
{
    public class HomeController : Controller
    {
        private MoviesDBEntities _entities = new MoviesDBEntities();

        public ActionResult Index()
        {
            var movies = _entities.MovieSet.ToList();
            return View("Index", movies);
        }

        [OutputCache(Duration = 15, VaryByParam = "None")]
        public ActionResult IndexCached()
        {
            var movies = _entities.MovieSet.ToList();
            return View("Index", movies);
        }

    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter11/VB/MvcApplication1/MvcApplication1/Controllers/HomeController.vb
<OutputCache(Duration:=30, VaryByParam:="None")> _
Public Class HomeController
    Inherits Controller

    Private _entities As New MoviesDBEntities()

    Public Function Index() As ActionResult
        Dim movies = _entities.MovieSet.ToList()
        Return View("Index", movies)
    End Function

    Public Function IndexCached() As ActionResult
        Dim movies = _entities.MovieSet.ToList()
        Return View("Index", movies)
    End Function

End Class
